cpp/include/raft/neighbors/ivf_pq_types.hpp

```diff
@@ -104,6 +104,25 @@ struct index_params : ann::index_params {
    * flag to `true` if you prefer to use as little GPU memory for the database as possible.
    */
   bool conservative_memory_allocation = false;
+
+  /**
+   * Helper that sets values according to the extents of the dataset mdspan.
```

Comment:
Can we please add usage guide here and potential update verbiage from "Helper that sets" to "Creates build params based on shape of input dataset"?